### PR TITLE
Add a simple clusterapi-tester, improve e2e logging.

### PR DESCRIPTION
**Pull request by bprashanth**
The first commit from https://github.com/kubernetes/kubernetes/pull/22253, so we can get the debug output ASAP without blocking on kubedns

### CODE REVIEW

File: test/e2e/util.go [hunk 1]
```diff
@@ -509,6 +511,40 @@ func waitForPodsRunningReady(ns string, minPods int, timeout time.Duration) erro
...
+func runKubernetesServiceTestContainer(repoRoot string, ns string) {
+	c, err := loadClient()
+	if err != nil {
+		Logf("Failed to load client")
+		return
+	}
+	p := podFromManifest(filepath.Join(repoRoot, "test", "images", "clusterapi-tester", "pod.yaml"))
+	p.Namespace = ns
+	_, err = c.Pods(ns).Create(p)
+	Expect(err).NotTo(HaveOccurred())
```
Comment (freehan) [inline test/e2e/util.go]: If error occurs here, will the whole e2e failed with this error? 
Or should we just log error here and continue with the "not all pods are ready" failure?

Same comment apply to all expect statement in this method...

Comment (bprashanth) [inline test/e2e/util.go]: Logf sounds like the right thing to do

**Comment by bprashanth** [comment]
PTAL

**Comment by bprashanth** [comment]
Marking this p0 because I really want some debug output for https://github.com/kubernetes/kubernetes/issues/21940

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please github issue: #22615

**Pull request merged**